lib/khttp/ktest/network.go

```diff
@@ -0,0 +1,45 @@
+// +build !release
+
+package ktest
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"net"
+)
+
+func AllocatePort() (*net.TCPAddr, error) {
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		return nil, err
+	}
+	allocatedDatastorePort, ok := listener.Addr().(*net.TCPAddr)
+	if !ok {
+		return nil, errors.New("shape of the address not correct, is your os not unix?")
+	}
+	return allocatedDatastorePort, nil
+}
+
+type KillAbleProcess []func()
+
+func (k *KillAbleProcess) KillAll() {
+	fmt.Println("running kill functions")
+	for _, killFunc := range *k {
+		fmt.Println("running kill function")
+		killFunc()
+	}
+}
+
+func (k *KillAbleProcess) AddKillable(process KillAbleProcess) {
+	if k == nil {
```

Comment:
do we need this if? when/why?

Reason: it can cause errors to go unnoticed, and necessary kill functions to not run. Eg, for some reason k is nil, the test is run, killable functions are added... but never run!!

I would remove the if, and let this fail in case we try to add a killable to a nil KillableProcess. That way, there is no risk that we end up with successful code ignoring necessary cleanups.

Reply:
We need this because it lets the testing library boil up killable funcs from other processes. and I agree, removed the if clause and just let it panic